verification/certifyKey.go

```diff
@@ -133,305 +31,50 @@ func TestCertifyKeySimulation(d TestDPEInstance, c DPEClient, t *testing.T) {
-	t.Helper()
-	var multiTcbInfo *TcgMultiTcbInfo
-	var err error
-
-	extKeyUsage := []asn1.ObjectIdentifier{}
-
-	for _, ext := range c.Extensions {
-		if ext.Id.Equal(OidExtensionExtKeyUsage) { // OID for ExtKeyUsage extension
-			// Extract the OID value from the extension
-			_, err := asn1.Unmarshal(ext.Value, &extKeyUsage)
-			if err != nil {
-				t.Errorf("[ERROR]: Failed to unmarshal the Extended Key Usage extension: %v", err)
-				continue
-			}
-
-			if !ext.Critical {
-				t.Errorf("[ERROR]: The Extended Key Usage extension IS NOT CRITICAL, MUST BE CRITICAL")
-			}
-			break
-		}
-	}
-
-	if len(extKeyUsage) == 0 {
-		t.Errorf("[ERROR]: The Extended Key Usage extension is empty")
-	}
-
-	// Iterate over the OIDs in the ExtKeyUsage extension
-	isExtendedKeyUsageValid := false
-	var expectedKeyUsage asn1.ObjectIdentifier
-	expectedKeyUsageName := ""
-	if c.IsCA {
-		expectedKeyUsage = OidExtensionTcgDiceKpEca
-		expectedKeyUsageName = "tcg-dice-kp-eca"
-	} else {
-		expectedKeyUsage = OidExtensionTcgDiceKpAttestLoc
-		expectedKeyUsageName = "tcg-dice-kp-attest-loc"
-	}
-
-	for _, oid := range extKeyUsage {
-		if oid.Equal(expectedKeyUsage) {
-			isExtendedKeyUsageValid = true
-			break
-		}
-	}
-	if !isExtendedKeyUsageValid {
-		t.Errorf("[ERROR]: Certificate has IsCA: %v  and does not contain specified key usage: %s", c.IsCA, expectedKeyUsageName)
-	}
-	return multiTcbInfo, err
-}
-
-// Checks for KeyUsage Extension as per spec
-// If IsCA = true, KeyUsage extension MUST contain DigitalSignature and KeyCertSign
-// If IsCA = false, KeyUsage extension MUST contain  only DigitalSignature
-func checkCertifyKeyExtensions(t *testing.T, c *x509.Certificate) {
-	t.Helper()
-
-	//Check for keyusage extension
-	var allowedKeyUsages x509.KeyUsage
-	if c.IsCA {
-		allowedKeyUsages = x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign
-	} else {
-		allowedKeyUsages = x509.KeyUsageDigitalSignature
-	}
-
-	certKeyUsageList := getKeyUsageNames(c.KeyUsage)
-	allowedKeyUsageList := getKeyUsageNames(allowedKeyUsages)
-	if c.KeyUsage != allowedKeyUsages {
-		t.Errorf("[ERROR]: Certificate has IsCA: %v and has got %v but want %v ", c.IsCA, certKeyUsageList, allowedKeyUsageList)
-	}
-
-}
-
-// Validates basic constraints in certificate returned by CertifyKey command
-// against the flag set for input parameter.
-// The BasicConstraints extension MUST be included
-// If CertifyKey AddIsCA is set, IsCA MUST be set to true.
-// If CertifyKey AddIsCA is NOT set, IsCA MUST be set to false
-func checkCertifyKeyBasicConstraints(t *testing.T, c *x509.Certificate, flags CertifyKeyFlags) {
-	t.Helper()
-
-	flagsBuf := &bytes.Buffer{}
-	binary.Write(flagsBuf, binary.LittleEndian, flags)
+// Checks whether FWID array omits index-1 when extend TCI is not supported in DPE profile.
+func TestCertifyKeyWithoutExtendTciSupport(d TestDPEInstance, c DPEClient, t *testing.T) {
+	simulation := false
```

Comment:
New test case to check if FWID[1] is omitted when no extendTCI support in DPE profile